packages/amplify-graphql-model-transformer/src/resources/rds-model-resource-generator.ts

```
@@ -23,92 +23,124 @@ import { ModelResourceGenerator } from './model-resource-generator';
-        'arn',
-        'aws',
-        'sns',
-        Fn.ref('AWS::Region'),
-        AmplifySQLLayerNotificationTopicAccount,
-        AmplifySQLLayerNotificationTopicName,
-      ]);
-
-      const patchingSubscriptionScope = context.stackManager.getScopeFor(SQLPatchingSubscriptionLogicalID, SQLStackName);
-      const snsTopic = Topic.fromTopicArn(patchingSubscriptionScope, SQLPatchingTopicLogicalID, topicArn);
-      const subscription = new LambdaSubscription(patchingLambda, {
-        filterPolicy: {
-          Region: SubscriptionFilter.stringFilter({
-            allowlist: [Fn.ref('AWS::Region')],
-          }),
-        },
-      });
-      snsTopic.addSubscription(subscription);
+    const strategies: Record<string, SQLLambdaModelDataSourceStrategy> = {};
+    if (strategyOverride) {
+      strategies[strategyOverride.name] = strategyOverride;
+    } else {
+      const dataSourceStrategies = Object.values(context.dataSourceStrategies).filter(isSqlStrategy);
+      dataSourceStrategies.forEach((strategy) => (strategies[strategy.name] = strategy));
+      const sqlDirectiveDataSourceStrategies = context.sqlDirectiveDataSourceStrategies?.map((dss) => dss.strategy) ?? [];
+      sqlDirectiveDataSourceStrategies.forEach((strategy) => (strategies[strategy.name] = strategy));
+    }
 
-      const lambdaDataSourceScope = context.stackManager.getScopeFor(SQLLambdaDataSourceLogicalID, SQLStackName);
-      const rdsDatasource = context.api.host.addLambdaDataSource(`${SQLLambdaDataSourceLogicalID}`, lambda, {}, lambdaDataSourceScope);
-      this.models.forEach((model) => {
-        context.dataSources.add(model, rdsDatasource);
-        this.datasourceMap[model.name.value] = rdsDatasource;
-      });
+    // Unexpected, since we invoke the generateResources in response to generators that are initialized during a scan of models and custom
+    // SQL, but we'll be defensive here.
+    if (Object.keys(strategies).length === 0) {
+      return;
```

Since we're biasing towards having the model's datasource strategy defined, if we hit this that means we missed something and we should throw an error.

---

Yeah, that's a good callout, will change.